document/paragraphproperties.go

```diff
@@ -199,3 +199,186 @@ func (p ParagraphProperties) SetHangingIndent(m measurement.Distance) {
+	if p.x.RPr.RStyle != nil {
+		return p.x.RPr.RStyle.ValAttr
+	}
+	return ""
+}
+
+// Font returns the name of paragraph font family.
+func (p ParagraphProperties) Font() string {
+	if fonts := p.x.RPr.RFonts; fonts != nil {
+		if fonts.AsciiAttr != nil {
+			return *fonts.AsciiAttr
+		} else if fonts.HAnsiAttr != nil {
+			return *fonts.HAnsiAttr
+		} else if fonts.CsAttr != nil {
+			return *fonts.CsAttr
+		}
+	}
+	return ""
+}
+
+// EastAsiaFont returns the name of paragraph font family for East Asia.
+func (p ParagraphProperties) EastAsiaFont() string {
+	if fonts := p.x.RPr.RFonts; fonts != nil {
+		if fonts.EastAsiaAttr != nil {
+			return *fonts.EastAsiaAttr
+		}
+	}
+	return ""
+}
+
+// ColorStr returns the hex value of paragraph color.
+func (p ParagraphProperties) ColorStr() string {
+	if color := p.x.RPr.Color; color != nil {
+		valAttr := color.ValAttr
+		if valAttr.ST_HexColorRGB != nil {
+			return *valAttr.ST_HexColorRGB
+		}
+	}
+	return ""
+}
+
+// SpacingValue returns the value of paragraph spacing.
+func (p ParagraphProperties) SpacingValue() int64 {
+	if spacing := p.x.RPr.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.Int64 != nil {
+			return *valAttr.Int64
+		}
+	}
+	return int64(0)
+}
+
+// SpacingMeasure returns the measure of paragraph spacing.
+func (p ParagraphProperties) SpacingMeasure() string {
+	if spacing := p.x.RPr.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.ST_UniversalMeasure != nil {
+			return *valAttr.ST_UniversalMeasure
+		}
+	}
+	return ""
+}
+
+// SizeValue returns the value of paragraph font size.
+func (p ParagraphProperties) SizeValue() float64 {
+	if sz := p.x.RPr.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// SizeMeasure returns the measure of paragraph font size.
+func (p ParagraphProperties) SizeMeasure() string {
+	if sz := p.x.RPr.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// ComplexSizeValue returns the value of paragraph font size for complex fonts.
+func (p ParagraphProperties) ComplexSizeValue() float64 {
+	if szCs := p.x.RPr.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// ComplexSizeMeasure returns the measure of paragraph font size for complex fonts.
+func (p ParagraphProperties) ComplexSizeMeasure() string {
+	if szCs := p.x.RPr.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// UnderlineType returns the type of paragraph underline.
+func (p ParagraphProperties) UnderlineType() wml.ST_Underline {
```

Comment:
Should we encapsulate wml.ST_Underline  into its own higher level type
with wml.ST_Underline accessible by its .X() function?  Might be more flexible for us.

Reply:
There is a function for RunProperties:
```go
func (r RunProperties) SetUnderline(style wml.ST_Underline, c color.Color) {
```
which uses the same type. Thus I think this is fine here.
Might be better to call it just ```Underline()``` ?

Reply:
so create a new struct with underline type and color as fields?